crates/bevy_ui/src/keyboard_navigation.rs

```diff
@@ -0,0 +1,173 @@
+use crate::{Interaction, UiStack};
+use bevy_a11y::Focus;
+use bevy_ecs::prelude::Component;
+use bevy_ecs::query::With;
+use bevy_ecs::system::{Local, Query, Res, ResMut};
+use bevy_ecs::{change_detection::DetectChangesMut, entity::Entity};
+use bevy_input::{prelude::KeyCode, Input};
+use bevy_reflect::{FromReflect, Reflect};
+use bevy_render::view::ComputedVisibility;
+
+/// A component that represents if a UI element is focused.
+#[derive(Reflect, FromReflect, Component, Clone, Debug, Default, Eq, PartialEq)]
+pub struct Focusable {
+    focus_state: FocusState,
+}
+
+impl Focusable {
+    /// The entity is currently focused, similar to the `:focus` css pseudo-class.
+    /// To check if the focus has been achieved through keyboard navigation, see [`Focusable::is_focus_visible`].
+    pub fn is_focused(&self) -> bool {
+        matches!(self.focus_state, FocusState::Focused { .. })
+    }
+
+    /// Focus has been reached through keyboard navigation and so a focus style should be displayed.
+    /// This is similar to the `:focus-visible` pseudo-class in css.
+    pub fn is_focus_visible(&self) -> bool {
+        matches!(self.focus_state, FocusState::Focused { visible: true })
+    }
+}
+
+#[derive(Reflect, FromReflect, Component, Clone, Debug, Default, Eq, PartialEq)]
```

Comment:
```suggestion
#[derive(Reflect, FromReflect, Clone, Debug, Default, Eq, PartialEq)]
```

This isn't a component.